mne_qt_browser/_pg_figure.py

```diff
@@ -4243,6 +4251,38 @@ def __init__(self, **kwargs):
             # disable histogram of epoch PTP amplitude
             del self.mne.keyboard_shortcuts["h"]
 
+        # Subscribe to vertical line change
+        subscribe(self, "time_change", self._on_time_change_event)
+
+        # Subscribe to time browse
+        subscribe(self, "time_browse", self._on_time_browse_event)
+
+        # Subscribe to channel browse
+        # self.mne.plt.sigYRangeChanged.connect(self._on_channel_browse_event)
+        subscribe(self, "channel_browse", self._on_channel_browse_event)
+
+    def _on_time_change_event(self, event):
+        """Response to TimeChange event from the event-ui system."""
+        with disable_ui_events(self):
+            self._add_vline(event.time)
+
+    def _on_time_browse_event(self, event):
+        """Response to TimeBrowse event from the event-ui system."""
+        with disable_ui_events(self):
+            self.mne.plt.setXRange(event.time_start, event.time_end, padding=0)
+
+    def _on_channel_browse_event(self, event):
+        """Response to ChannelBrowse event from the event-ui system."""
+        # Get the indices of the subset in the full set of channels
+        all_channels = self.mne.ch_names[self.mne.ch_order]
+        ch_indices = [np.where(all_channels == ch)[0][0] for ch in event.channels]
+
+        # Take the start index and set range
+        with disable_ui_events(self):
+            start_idx = ch_indices[0]
+            n_chans = len(ch_indices)
+            self.mne.plt.setYRange(start_idx, start_idx + n_chans + 1, padding=0)
```

Comment:
This assumes the desired channels (`event.channels`) are all right next to each other. How about `start_idx, end_idx = ch_indices.min(), ch_indices.max()`?